src/machine.rs

```diff
@@ -270,6 +270,38 @@ impl<'m> Machine<'m> {
         Ok(())
     }
 
+    /// Delete the machine.
+    ///
+    /// Deletes the machine, cleaning up all associated resources.
+    ///
+    /// Machine is shutdown before resources are deleted.
+    #[instrument(skip_all)]
+    pub async fn delete(mut self) -> Result<(), Error> {
+        let vm_id = self.config.vm_id().to_string();
+        let jailer_workspace_dir = self.config.jailer_workspace_dir.to_owned();
+
+        if let Err(err) = self.shutdown().await {
+            trace!("{vm_id}: Shutdown error: {err}");
+        }
+
+        info!("{vm_id}: Waiting for the VM process to shut down...");
+        sleep(Duration::from_secs(10)).await;
+
+        if let Err(err) = self.force_shutdown().await {
+            trace!("{vm_id}: Forced shutdown error: {err}");
+        }
```

Comment:
* With #17 merged, we should only shutdown if the state is RUNNING.
* If shutdown() call was unsuccessful, we shouldn't wait for machine to shutdown and proceed with killing it.